metaflow/plugins/kfp/tests/run_integration_tests.py

```diff
@@ -74,6 +79,70 @@ def test_raise_failure_flow(pytestconfig) -> None:
     return
 
 
+def validate_node_selector_term(node_selector_term: Dict) -> bool:
+    for affinity_match_expression in node_selector_term["matchExpressions"]:
+        if (
+            affinity_match_expression["key"] == "k8s.amazonaws.com/accelerator"
+            and affinity_match_expression["operator"] == "In"
+            and "nvidia-tesla-v100" in affinity_match_expression["values"]
+        ):
+            return True
+    return False
+
+
+def validate_toleration(toleration: Dict) -> bool:
+    if (
+        toleration["effect"] == "NoSchedule"
+        and toleration["key"] == "k8s.amazonaws.com/accelerator"
+        and toleration["operator"] == "Equal"
+        and toleration["value"] == "nvidia-tesla-v100"
+    ):
+        return True
+    return False
+
+
+def test_compile_only_accelerator_test() -> None:
+    compile_to_yaml_cmd = (
+        f"{_python()} flows/accelerator_flow.py --datastore=s3 kfp run "
+        f" --no-s3-code-package --yaml-only --pipeline-path accelerator_flow.yaml"
+    )
+
+    compile_to_yaml_process = run(
+        compile_to_yaml_cmd,
+        universal_newlines=True,
+        stdout=PIPE,
+        shell=True,
+    )
+    assert compile_to_yaml_process.returncode == 0
+
+    with open("accelerator_flow.yaml", "r") as stream:
+        try:
+            flow_yaml = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    for step in flow_yaml["spec"]["templates"]:
+        if step["name"] == "start":
+            start_step = step
+            break
+    else:
+        raise Exception("Start step could not be found.")
+
+    for node_selector_term in start_step["affinity"]["nodeAffinity"][
+        "requiredDuringSchedulingIgnoredDuringExecution"
+    ]["nodeSelectorTerms"]:
+        if validate_node_selector_term(node_selector_term):
+            break
+    else:
+        raise Exception("Correct node affinity for P3 instance type not found.")
```

Comment:
You should change this as an assertion or test validation rather than raising an Exception, same for the other raising of Exception.